### PR TITLE
Update code obfuscation for Android / iOS

### PR DESCRIPTION
**Pull request by sushi2k**
Thank you for submitting a Pull Request to the Mobile Security Testing Guide. Please make sure that:

- [ ] Your contribution is written in the 2nd person (e.g. you)
- [ ] Your contribution is written in an active present form for as much as possible.
- [ ] You have made sure that the reference section is up to date (e.g. please add sources you have used, make sure that the references to MITRE/MASVS/etc. are up to date)
- [ ] Your contribution has proper formatted markdown and/or code
- [ ] Any references to website have been formatted as [TEXT](URL “NAME”)
- [ ] You verified/tested the effectiveness of your contribution (e.g.: is the code really an effective remediation? Please verify it works!)

If your PR is related to an issue. Please end your PR test with the following line:
This PR covers issue #1116 .

### CODE REVIEW

**Comment by OMTGreleaser** [comment]
Results for commit da086eac05198cd1a333f50073cda9c9fec38a4f: Broken link result: [!] Error(s) found by the Links validator: 9 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:       52 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

**Comment by OMTGreleaser** [comment]
Results for commit 47c2b04287af48eb88966aa2b8312be53769d55c: Broken link result: [!] Error(s) found by the Links validator: 9 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:       53 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

File: Document/0x06j-Testing-Resiliency-Against-Reverse-Engineering.md [hunk 1]
```diff
@@ -607,6 +607,88 @@ The goal of emulator detection is to increase the difficulty of running the app
...
+
+In the original source code you can see all the class and method identifiers:
+
+<img src="Images/Chapters/0x06j/no_obfuscation.png" width="550px"/>
+
+SwiftShield was now replacing all of them with encrypted values that leave no trace to their original name or intention of the class/method:
+
+<img src="Images/Chapters/0x06j/swiftshield_obfuscated.png" width="650px"/>
+
+After executing `swiftshield` a new directory will be created called `swiftshield-output`. In this directory another directory is created with a time stamp in the folder name. This directory contains a text file called `conversionMap.txt`, that maps the encrypted strings to it's original values.
```
Comment (cpholguera) [inline Document/0x06j-Testing-Resiliency-Against-Reverse-Engineering.md]:
```suggestion
After executing `swiftshield` a new directory will be created called `swiftshield-output`. In this directory another directory is created with a timestamp in the folder name. This directory contains a text file called `conversionMap.txt`, that maps the encrypted strings to their original values.
```

File: Document/0x06j-Testing-Resiliency-Against-Reverse-Engineering.md [hunk 2]
```diff
@@ -607,6 +607,88 @@ The goal of emulator detection is to increase the difficulty of running the app
...
+viewDidLoad ===> DLaNRaFbfmdTDuJCPFXrGhsWhoQyKLnO
+sceneDidBecomeActive ===> SUANAnWpkyaIWlGUqwXitCoQSYeVilGe
+AppDelegate ===> KftEWsJcctNEmGuvwZGPbusIxEFOVcIb
+Deny_Debugger ===> lKEITOpOvLWCFgSCKZdUtpuqiwlvxSjx
+Button_Emulator ===> akcVscrZFdBBYqYrcmhhyXAevNdXOKeG
+```
+
+This is needed for [deobfuscating encrypted crash logs](https://github.com/rockbruno/swiftshield#-deobfuscating-encrypted-crash-logs "Deobfuscating encrypted Crash logs").
+
+Another example project is available in the [Github repo](https://github.com/rockbruno/swiftshield/tree/master/SwiftShieldExample "SwiftShieldExample"), that can be used to test the execution of SwiftShield.
```
Comment (cpholguera) [inline Document/0x06j-Testing-Resiliency-Against-Reverse-Engineering.md]:
```suggestion
Another example project is available in SwiftShield's [Github repo](https://github.com/rockbruno/swiftshield/tree/master/SwiftShieldExample "SwiftShieldExample"), that can be used to test the execution of SwiftShield.
```

File: Document/0x06j-Testing-Resiliency-Against-Reverse-Engineering.md [hunk 3]
```diff
@@ -607,6 +607,88 @@ The goal of emulator detection is to increase the difficulty of running the app
...
+
+#### Effectiveness Assessment
+
+Attempt to disassemble the Mach-O in the IPA and any included library files in the "Frameworks" directory (.dylib or .framework files), and perform static analysis. At the very least, the app's core functionality (i.e., the functionality meant to be obfuscated) shouldn't be easily discerned. Verify that
```
Comment (cpholguera) [inline Document/0x06j-Testing-Resiliency-Against-Reverse-Engineering.md]:
```suggestion
Attempt to disassemble the Mach-O in the IPA and any included library files in the "Frameworks" directory (.dylib or .framework files), and perform static analysis. At the very least, the app's core functionality (i.e., the functionality meant to be obfuscated) shouldn't be easily discerned. Verify that:
```

File: Document/0x06j-Testing-Resiliency-Against-Reverse-Engineering.md [hunk 4]
```diff
@@ -607,6 +607,88 @@ The goal of emulator detection is to increase the difficulty of running the app
...
+
+- meaningful identifiers, such as class names, method names, and variable names, have been discarded,
+- string resources and strings in binaries are encrypted,
```
Comment (cpholguera) [inline Document/0x06j-Testing-Resiliency-Against-Reverse-Engineering.md]:
```suggestion
- string resources and strings in binaries are encrypted.
```

File: Document/0x06j-Testing-Resiliency-Against-Reverse-Engineering.md [hunk 5]
```diff
@@ -607,6 +607,88 @@ The goal of emulator detection is to increase the difficulty of running the app
...
+
+This is needed for [deobfuscating encrypted crash logs](https://github.com/rockbruno/swiftshield#-deobfuscating-encrypted-crash-logs "Deobfuscating encrypted Crash logs").
+
+Another example project is available in the [Github repo](https://github.com/rockbruno/swiftshield/tree/master/SwiftShieldExample "SwiftShieldExample"), that can be used to test the execution of SwiftShield.
+
+#### Effectiveness Assessment
+
+Attempt to disassemble the Mach-O in the IPA and any included library files in the "Frameworks" directory (.dylib or .framework files), and perform static analysis. At the very least, the app's core functionality (i.e., the functionality meant to be obfuscated) shouldn't be easily discerned. Verify that
+
+- meaningful identifiers, such as class names, method names, and variable names, have been discarded,
```
Comment (cpholguera) [inline Document/0x06j-Testing-Resiliency-Against-Reverse-Engineering.md]:
```suggestion
- meaningful identifiers, such as class names, method names, and variable names, have been discarded.
```

**Comment by sushi2k** [comment]
Thanks @cpholguera. Applied all suggestions, please have a look again!

**Comment by OMTGreleaser** [comment]
Results for commit 9998716183daa10dce75fe1be690d3978c5ec9fa: Broken link result: [!] Error(s) found by the Links validator: 10 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:        9 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

**Pull request merged**